sstables/sstables.cc

```diff
@@ -3320,8 +3320,63 @@ gc_clock::time_point sstable::get_gc_before_for_fully_expire(const gc_clock::tim
     return res.knows_entire_range ? res.min_gc_before : gc_clock::time_point::min();
 }
 
+// Returns error code, 0 is success
+static future<int> remove_dir(fs::path dir, bool recursive) {
+    try {
+        co_await (recursive ? recursive_remove_directory(dir) : remove_file(dir.native()));
+        co_return 0;
+    } catch (...) {
+        auto ex = std::current_exception();
+        int error_code = -1;
+        try {
+            std::rethrow_exception(ex);
+        } catch (const std::system_error& e) {
```

Comment:
Minor, but you can just catch std::system_error in the outer catch.

Reply:
Done in v4